C/typeSkipTest.c

```diff
@@ -0,0 +1,62 @@
+#include "typeSkipTest.h"
+
+/* A length-prefixed encoding of the following Simplicity program:
+ *     witness (runIdentity (getValue (return True))) >>> mn >>> unit
+ *      where
+ *       l1 = take l0 &&& drop l0
+ *       l2 = take l1 &&& drop l1
+ *       l3 = take l2 &&& drop l2
+ *       ltop = l3
+ *       m1 = copair l3 l3
+ *       m2 = take l1 &&& drop m1
+ *       m3 = take m2 &&& drop l2
+ *       m4 = take l3 &&& drop m3
+ *       m5 = copair (injl m4) (injr ltop)
+ *       m6 = take l1 &&& drop m5
+ *       m7 = take m6 &&& drop l2
+ *       m8 = take l3 &&& drop m7
+ *       n1 = copair l3 l3
+ *       n2 = take n1 &&& drop l1
+ *       n3 = take l2 &&& drop n2
+ *       n4 = take n3 &&& drop l3
+ *       n5 = copair (injl ltop) (injr n4)
+ *       n6 = take n5 &&& drop l0
+ *       n7 = take l1 &&& drop n6
+ *       n8 = take n7 &&& drop l2
+ *       mn = copair (injl m8) (injr n8)
```

Comment:
This looks like an optimized version of a [similar test case in asset-gen](https://github.com/uncomputable/asset-gen/blob/aad0f90ad7ff066c4ab54bb2015767e4c21939b9/src/main.rs#L46). Out of curiosity, could you explain what the program in this PR does that the program in asset-gen doesn't do?

Reply:
I consider this test different than the one in asset-gen.

The one in asset-gen is trying to illustrate exponential time savings by giving a case that runs fast when type skip is implemented and would otherwise run slow.

The test here is trying to extend code coverage by hitting various branches in `type.h`'s `typeSkip`, `dag.c`'s `fillWitnessData`, and `eval.c`'s `writeValue`.